app/src/main/java/org/grapheneos/apps/client/utils/NotificationManager.kt

```diff
@@ -0,0 +1,103 @@
+        App.INSTALLATION_FAILED_GROUP,
+        context.getString(R.string.installationFailed)
+    )
+    listOf(
+        seamlessUpdateGroup,
+        installationFailedGroup
+    ).forEach {
+        createNotificationChannelGroup(it)
+    }
+
+    val installationFailed = NotificationChannelCompat.Builder(
+        App.INSTALLATION_FAILED_CHANNEL,
+        NotificationManager.IMPORTANCE_HIGH
+    ).setName(context.getString(R.string.installationFailed))
+        .setVibrationEnabled(false)
+        .setGroup(App.INSTALLATION_FAILED_GROUP)
+        .setShowBadge(false)
+        .setLightsEnabled(false)
+        .build()
+
+    val channelSeamlesslyUpdated = NotificationChannelCompat.Builder(
+        App.SEAMLESSLY_UPDATED_CHANNEL,
+        NotificationManager.IMPORTANCE_LOW
+    ).setName(context.getString(R.string.nUpdatedTitle))
+        .setDescription(context.getString(R.string.nUpdatedDescription))
+        .setVibrationEnabled(false)
+        .setGroup(App.SEAMLESS_UPDATE_GROUP)
+        .setShowBadge(false)
+        .setLightsEnabled(false)
+        .build()
+
+    val channelConformationNeeded = NotificationChannelCompat.Builder(
+        App.SEAMLESS_UPDATE_INPUT_REQUIRED_CHANNEL,
+        NotificationManager.IMPORTANCE_HIGH
+    ).setName(context.getString(R.string.nUpdateAvailableTitle))
+        .setDescription(context.getString(R.string.nUpdateAvailableDescription))
+        .setVibrationEnabled(false)
+        .setGroup(App.SEAMLESS_UPDATE_GROUP)
+        .setShowBadge(false)
+        .setLightsEnabled(false)
+        .build()
+
+    val channelAlreadyUpToDate = NotificationChannelCompat.Builder(
+        App.ALREADY_UP_TO_DATE_CHANNEL,
+        NotificationManager.IMPORTANCE_MIN
+    ).setName(context.getString(R.string.nUpToDateTitle))
+        .setDescription(context.getString(R.string.nUpToDateDescription))
+        .setVibrationEnabled(false)
+        .setGroup(App.SEAMLESS_UPDATE_GROUP)
+        .setShowBadge(false)
+        .setLightsEnabled(false)
+        .build()
+
+    val channelSeamlessUpdateFailed = NotificationChannelCompat.Builder(
+        App.SEAMLESS_UPDATE_FAILED_CHANNEL,
+        NotificationManager.IMPORTANCE_DEFAULT
+    ).setName(context.getString(R.string.nUpdatesFailedTitle))
+        .setDescription(context.getString(R.string.nUpdatesFailedDescription))
+        .setVibrationEnabled(false)
+        .setShowBadge(false)
+        .setGroup(App.SEAMLESS_UPDATE_GROUP)
+        .setLightsEnabled(false)
+        .build()
+
+    val channelBackgroundTask = NotificationChannelCompat.Builder(
+        App.BACKGROUND_SERVICE_CHANNEL,
+        NotificationManager.IMPORTANCE_LOW
+    ).setName(context.getString(R.string.nBackgroundTaskTitle))
+        .setDescription(context.getString(R.string.nBackgroundTaskDescription))
+        .setVibrationEnabled(false)
+        .setShowBadge(false)
+        .setLightsEnabled(false)
+        .build()
+
+    listOf(
```

Comment:
`createNotificationChannels()` allows to create multiple notification channels with a single call to system_server.

Reply:
`public void createNotificationChannels(@NonNull List<NotificationChannel> channels)` it doesn't work for `NotificationChannelCompat`

Reply:
Why was `NotificationChannelCompat` used instead of `NotificationChannel`?

Reply:
And there's `createNotificationChannelsCompat`: https://developer.android.com/reference/androidx/core/app/NotificationManagerCompat#createNotificationChannelsCompat(java.util.List%3Candroidx.core.app.NotificationChannelCompat%3E)

Reply:
I don't recall correctly.